### ISSUE TITLE
Is there a prebuilt shared library?

### ISSUE THREAD

**Issue by CodeMyst** [go-enry/go-enry#18]
I couldn't find any way in this repo on how to get the shared library so it can be used from other languages.

So is there a prebuilt one somewhere or some instructions on how to build your own?

**Comment by erizocosmico** [comment]
You can generate a shared library like this:

**For linux:**

```
make linux-shared
```

**For macOS:**

```
make darwin-shared
```

Note that for generating the linux shared you need to be on a linux system and for the darwin shared image you need to be on a darwin system. You can generate the darwin shared library from linux if you have cross compilation set up. [The travis yaml might help you with that](https://github.com/go-enry/go-enry/blob/master/.travis.yml#L87-L108).

**Comment by CodeMyst** [comment]
Ah thanks, how about generating a lib for windows?

also this make stuff should probably be put somewhere in readme.

**Comment by erizocosmico** [comment]
Previously the shared libraries were uploaded to each release, but the repository move and change to GitHub actions the part of the travis config file that was in charge of that didn't get ported. I raised an issue about this at https://github.com/go-enry/go-enry/issues/19.

At the time when this feature was done c-shared was still not implemented on windows. It was implemented some months later, but this never got revisited in enry. I will add an issue about this as well.
